analyses/cell-type-ewings/cell_type_annotations/SCPCS000490/SCPCL000822_tumor-cell-validation.Rmd

```diff
@@ -0,0 +1,600 @@
+Any cells with a total marker gene expression value > 0 were called as tumor cells. 
+- CopyKAT - no reference: [`CopyKAT`](https://github.com/navinlabcode/copykat) was run without indicating any reference cells and cells were classified as aneuploid (Tumor) or diploid (Normal). 
+- InferCNV - [`InferCNV`](https://github.com/broadinstitute/inferCNV/wiki) was used to call copy number variations using endothelial and fibroblasts assigned by `CellAssign` as the normal reference group. 
+This returns a proportion of each chromosome with a CNV detected.
+We then calculate the genomic CNV proportion for each cell across all chromosomes, weighted by the number of genes in a chromosome.
+Cells with a genomic CNV proportion greater than the mean  CNV proportion across all cells are called as tumor cells.
+
+
+```{r}
+# plot tumor cells for each classification method 
+tumor_umap_df <- classification_df |> 
+  dplyr::select(barcodes, UMAP1, UMAP2, marker_gene_classification, copykat_classification, cnv_proportion_classification) |> 
+  tidyr::pivot_longer(
+    cols = ends_with("classification"),
+    names_to = "method",
+    values_to = "classification"
+  ) |> 
+  dplyr::mutate(
+    method = dplyr::case_when(
+      method == "marker_gene_classification" ~ "Marker genes only",
+      method == "copykat_classification" ~ "CopyKAT - no reference",
+      method == "cnv_proportion_classification" ~ "InferCNV proportion"
+    )
+  )
+
+ggplot(tumor_umap_df, aes(x = UMAP1, UMAP2, color = classification)) +
+  geom_point(size = 0.5, alpha = 0.5) +
+  facet_wrap(vars(method))
+```
+
+This plot shows us that there is a pretty clear consensus that most of the cells in the bottom right group are tumor cells. 
+From the original marker gene analysis we know that not all marker genes are expressed and that there is quite a bit of variance in expression. 
+Because of that and the fact that both CNV inference methods show similar annotations, we will classify any cells that are labeled as tumor by both `CopyKAT` and `InferCNV` as tumor cells. 
+
+```{r}
+# create a combined classification between inferCNV and copyKAT
+# cells that are classified as tumor by both are kept as tumor
+# classified by both as normal are normal 
+# in between are classified as unknown
+classification_df <- classification_df |> 
+  dplyr::mutate(
+    combined_cnv_classification = dplyr::case_when(
+      # only tumor if both call tumor 
+      cnv_proportion_classification == "Tumor" & copykat_classification == "Tumor" ~ "Tumor",
+      # only normal if both call normal and/or reference
+      cnv_proportion_classification == "Normal" & copykat_classification == "Normal" ~ "Normal",
+      cnv_proportion_classification == "Reference" & copykat_classification == "Normal" ~ "Normal",
+      # if normal/tumor mismatch set to unknown 
+      .default = "Unknown"
```

Comment:
I don't really like "Unknown" here. Maybe something more descriptive?
I suggested "conflicted", but "ambiguous", "disagreement" or something like that?

You may also want to throw in a factor ordering step here?

```suggestion
      # if normal/tumor mismatch set to Conflicted
      .default = "Conflicted"
```